### PR TITLE
Fix stopping dev server on Windows 

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Starting killing the dev server process on Windows since sending signals (and SIGTERM or os.Interrupt) is not supported 

## Why?

Ensuring the dev server is stopped and doesn't leave hanging

## Checklist

1. Closes 

2. How was this tested:

Added test

3. Any docs updates needed?

### CODE REVIEW

**Review by bergundy — COMMENTED** [review]
See how to interrupt a process in windows:

https://github.com/temporalio/omes/blob/main/cmd/util_windows.go

**Comment by feedmeapples** [comment]
> See how to interrupt a process in windows:
> 
> https://github.com/temporalio/omes/blob/main/cmd/util_windows.go

added here! Also not sure if it is actually meant to be interrupt in the `omes` repo. I've replaced this with `SIGTERM` in dev server

**Comment by feedmeapples** [comment]
having unrelated issue with sdk-go v1.23.0. Recreated the PR on top of 1.21.2 until fixing CLI's CI with newer versions

File: testsuite/devserver.go [hunk 1]
```diff
@@ -82,7 +81,7 @@ type DevServerOptions struct {
 
 // Temporal CLI based DevServer
 type DevServer struct {
-	cmd    *exec.Cmd
+	Cmd    *exec.Cmd
```
Comment (feedmeapples) [inline testsuite/devserver.go:84]: not a fan of making it public, though this seems to be the only option to test the RC after interrupting the server (especially for Windows)

Comment (feedmeapples) [reply testsuite/devserver.go:84]: unless we don't test this and only rely on `/cli`'s tests

**Comment by cretz** [comment]
> having unrelated issue with sdk-go v1.23.0

Make sure the `go.temporal.io/api` and `go.temporal.io/sdk` are _both_ on the exact same versions as the `go.mod` of the server version you are using. Even if it compiles, there can be other issues if you mismatch. Server version should decide the other two.

File: testsuite/interrupt_windows.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
+//go:build windows
```
Comment (cretz) [inline testsuite/interrupt_windows.go:1]: This is not needed, it's assumed by the `_windows.go` filename

File: testsuite/interrupt_windows.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"os"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+)
+
+// sendTerminate calls the break event on the given process for graceful shutdown.
+func sendTerminate(process *os.Process) error {
```
[the remark this answers is not in the document]
Comment (feedmeapples) [reply testsuite/interrupt_windows.go]: > > having unrelated issue with sdk-go v1.23.0
> 
> Make sure the `go.temporal.io/api` and `go.temporal.io/sdk` are _both_ on the exact same versions as the `go.mod` of the server version you are using. Even if it compiles, there can be other issues if you mismatch. Server version should decide the other two.

changed back to the `master` branch

**Review by cretz — APPROVED** [review]
Unfortunately we aren't running CI in Windows here like other ones, but LGTM and I am just hoping it works

**Pull request merged**